packages/compass-sidebar/src/components/legacy/sidebar-databases-navigation.tsx

```diff
@@ -348,6 +348,8 @@ const mapStateToProps: MapStateToProps<
         connectionStatus: ConnectionStatus.Connected,
         isPerformanceTabSupported:
           !isDataLake && !!state.isPerformanceTabSupported[connectionId],
+        isGenuineMongoDB: instance?.genuineMongoDB.isGenuine !== false,
```

Comment:
this feels a tiny bit unsafe, as it really means "is not a confirmed non genuine". most likely not an issue, but I imagined that in the future someone would say "let's show a badge that it's genuine!" and we'd use this bool which is true for undefined.

Reply:
I think it gets more clearer when we look at how this (genuineMongoDB) is [computed](https://github.com/mongodb-js/mongodb-build-info/blob/main/index.js#L90-L110) because it is set to false only for explicitly identified conditions. We might as well say that every connection is genuine unless it was explicitly identified not to be.

Reply:
ah I see. so looks like this can only be other than a boolean when there is no instance. which shouldn't even happen, feels like we're missing an if somewhere :D